reg/drone/srv/air_data_computer/_.0.1.uavcan

```diff
@@ -0,0 +1,48 @@
+# A generic air data computer (ADC) service.
+#
+# An air data computer service is generally a non-interactive publish-only service: when activated,
+# it keeps publishing its measurements at a fixed rate over several subjects until deactivated.
+# The activation/deactivation, if supported, is managed via the standard arming command service.
+# The data update rates and other parameters are controlled via the Register API using implementation-defined
+# register names.
+#
+# An ADC whose arming state is SLEEP is allowed to cease all network activity.
+#
+# An ADC whose arming state is DISARMED is recommended to behave as if its state was ARMED;
+# implementations are recommended to leverage this state to perform any necessary maintenance activities
+# such as calibration, self-heating, etc. The ADC should not report its own status as ARMED until said
+# maintenance activities have been completed. Therefore, the high-level controler (e.g., the flight
+# management unit) may delay the take-off until the ADC is ready.
+#
+# An ADC whose arming state is ARMED is required to comply with the requirements set out below.
+#
+# An armed ADC should publish the following data subjects synchronously at the same configurable rate
+# which should not be lower than the specified limit. The measurements should be low-pass filtered
+# to avoid frequency aliasing effects.
+#
+#   PUBLISHED SUBJECT       TYPE                                                NOTE
+#   calibrated airspeed     reg.drone.phy.kinematics.trans.Velocity1CovTs
+#   true airspeed           reg.drone.phy.kinematics.trans.Velocity1CovTs       optional
+#   pressure altitude data  reg.drone.phy.kinematics.trans.LinearCovTs          assume ISA; NED frame (+down, -up)
+#   static air data         reg.drone.phy.thermodynamics.PressureTempCovTs      pressure and OAT
+#
+# Observe that there is no subject for indicated airspeed. This is because per the principles of service-oriented
+# design, the ADC is responsible for applying the necessary transformations on its data to render it ready for
+# consumption by clients. If desired, indicated airspeed may be reported over an implementation-defined subject.
```

Comment:
I am not sure about this. I will request to an SME if this is actually something expected to come out of the sensor or not. I understand we need to follow the service-oriented design, but without proper extension of the service these data types are unusable.

Reply:
@bresch can I have your comment here?

Reply:
My question is "what do we expect from a sensor that cannot compute CAS"? Should it simply publish IAS in the CAS field? How the client would then know that it should apply calibration on that data?

Reply:
> My question is "what do we expect from a sensor that cannot compute CAS"? Should it simply publish IAS in the CAS field? How the client would then know that it should apply calibration on that data?

The short answer is that if a sensor is unable to produce highly processed data that is abstracted away from the means of its estimation, it is non-conformant. This is a very conscious requirement; we need to keep in mind that this standard is intended to enable distributed intravehicular systems and it is not a bad clone of I2C. The long answer is provided in [The Guide](uavcan.org/guide/).

The vendor of a [COTS air data computer](https://www.tindie.com/products/avionicsanonymous/uavcan-air-data-computer-airspeed-sensor/) would provide configuration parameters (see Register API) that define the relationship between IAS and CAS. The parameters are to be configured by the integrator based on the position of the pitot sensor and other relevant properties.

Reply:
> The short answer is that if a sensor is unable to produce highly processed data that is abstracted away from the means of its estimation, it is non-conformant. This is a very conscious requirement; we need to keep in mind that this standard is intended to enable distributed intravehicular systems and it is not a bad clone of I2C.

@pavel-kirienko so we just assume that a vendor needs to build a sensor which in the end will be much more expensive because it can provide CAS? I mean, I don't think this is a reasonable way of seeing things. DS-015 is aimed for a board spec of solutions, from the less to the most expensive. It's more expensive to create a COTS air computer like that than providing a field for CAS.

Reply:
IAS-CAS translation does not require any additional hardware, but a trivial piecewise transfer function, so I don't think the cost is affected. If the cost is really critical, I2C is the way to go, otherwise you're going against the core design goals of the standard.

Reply:
> otherwise you're going against the core design goals of the standard.

What exactly are we going against? One of the core design goals of the standard is to allow flexibility for the HW vendor to specify what sort of data it wants to provide.

Reply:
The goals are defined in the MVP doc in the form of a reference to the interface design guidelines. The values announced there are, therefore, the core values of DS-015. We are venturing too far away from the scope of this PR here with this discussion.

> The main motivation for our (Zubax) involvement in this project is that we perceive that the early draft of the standard proposed for public evaluation is not built according to the best practices of diligent design and it is not an adequate candidate for a long-term industry standard. Given the strategic importance of the standard both for the PX4 ecosystem and for the core UAVCAN project itself, leaving the project on its current trajectory may reflect poorly on the future of the entire industry. Our criticism of the proposed design is expressed in the form of advisory guidance in the freshly published chapter on Interface Design Guidelines. Despite the design guidance being available, it does not appear to be sufficient to steer the design process off its current trajectory, so we choose to intervene directly.
>
> To implement the work described in this document we request authority to override design decisions made by the Auterion team.